### DISCUSSION TITLE
Heat Transfer Through Opening (Excessive Temperatures Results of Upper Layer Temperature)

### DISCUSSION THREAD

**Discussion by SXFire12** [firemodels/cfast#2125] [Q&A] [answered]
Dear All, 
I'm trying to compare the results from the CFAST and FDS, however, the CFAST results in terms of upper layer temperature tend to have higher values than the FDS, especially in rooms 8 and 27 (the picture is given). I tried to simulate some variances of the HRR ranging from 150 kW to 3000 kW to see the pattern and compare it between CFAST and FDS as attached. 

I found out the previous discussion 
https://groups.google.com/g/cfast/c/9hmJNF5jdmc (excessive temperature due to adiabatic compartment surfaces and/or no heat loss through the window)

However, in my case more into the temperatures in the adjacent rooms. Do you think this is due to the ratio of the compartments? (L/H and W/H)? Or is there any reason for that? 

Thanks 

![Compartments](https://github.com/user-attachments/assets/7a9208ea-903c-4df2-970b-fae67504d510)
![All Simulation](https://github.com/user-attachments/assets/74bfc27f-d08e-4a7e-b4bf-37293995a5a8)

[FDS_CFAST.zip](https://github.com/user-attachments/files/16247797/FDS_CFAST.zip)

**Comment by drjfloyd** [comment, marked as answer]
FDS and CFAST are very different codes.  CFAST for example cannot account for the fact that the room 27 doorway is offset in the corner diagonally opposite from where rooms 6 and 8 are. But there are also many differences between your inputs and potential issues with your FDS inputs:

-You don't have great grid resolution for the two small room doorways.  The 6 door is 2 cells wide and 8 is 3 cells.
-Your door openings are not the same size for CFAST and FDS. For example, in CFAST 6-27 is 0.72 x 1.77 and in FDS on your grid it is 0.53 x 1.82. These are A*H^0.5 of 1.7 and 1.3 which is a fairly significant difference.
-In CFAST th C-Class surface is 1 cm thick and in FDS you have it 6 cm thick.  In CFAST you use both C-Class and A-60 class and in FDS you only have C-Class for the inner compartments and for the outer compartment you have not specified a boundary condition and are getting the default FDS boundary condition which is a fixed 20 C wall (so very different from the CFAST model).
-The FDS layer height device only has meaning if you place it in a location where a layer is expected. In room 6 you have placed the device over the fire, so you are in the plume.  In room 27, given the assymetric layout, you may not be developing a symmetric layer. You should look at multiple layer height locations to see if there is a variance in FDS.
-You have a grid aspect ratio of almost 3:1.  This is not recommended in the FDS User's Guide.

**Reply by SXFire12** [reply]
Dear Dr. Floyd, thanks for all of your feedback. In terms of ratio, I have done another simulation with more complex geometries and with a ratio of 1:8, and gave me significant results as well. However, with my computational resources, it took me 1 day to resolve it. That´s why I put a coarser mesh for these simulations just to see the variations of HRR with the generated temperature, I was thinking maybe there are some HRR value limitations for the generated temperature as CFAST uses a Heskestad’s correlation for the plume temperature. 

And why we cannot account for the room 27 doorway is offset in the corner diagonally opposite from where rooms 6 and 8 are?

And for room 27 I put several devices in FDS, and average it to compare with the CFAST

**Reply by drjfloyd** [reply]
CFAST is a zone model. It doesn't have detailed geometry. CFAST only knows the compartment floor area, the compartment height, and the area of vents.

**Reply by SXFire12** [reply]
Sorry, maybe I should revise my words, not complex geometries, but more compartments I would say,

**Reply by drjfloyd** [reply]
CFAST is a very simple model. You cannot expect it to have the same performance for equivalent inputs as FDS for more complex geometries. Your biggest issue in the plots you attached is probably using INERT in FDS (you did not defined boundary conditions for all solid surfaces). This error will result in FDS underpredicting temperatures. 

FDS is a CFD code. The accuracy of its results is tied to the grid resolution. Having reasonably well resolved simulations take more than a day is not unusual with FDS. It is not CFAST. It is not intended to run in minutes for typical problems.

**Reply by SXFire12** [reply]
Thanks, Dr. Floyd for your feedback. I re-run my model again to confront the defined boundary

**Comment by SXFire12** [comment]
I have a curiosity about this statement " CFAST for example cannot account for the fact that the room 27 doorway is offset in the corner diagonally opposite from where rooms 6 and 8 are". Why that is the case? I did not find the answer in the manual for the vent position. Could somebody explain it? Thanks

**Reply by drjfloyd** [reply]
CFAST is a zone model. A room is just two points: the upper layer and the lower layer. There is no north, south, east, and west for vent flow. It is just in or out of a layer. All CFAST has for solving for the door flow is the layer height and temperature and the compartment pressure. It does not account for anything to do with what wall or where on the wall a door is.

**Reply by SXFire12** [reply]
does it correct to create one room with an uneven compartment by using a vent as shown below? 
![Captura de pantalla 2025-04-07 184906](https://github.com/user-attachments/assets/14f7cc64-6ed4-499e-be53-e7f756a64504)

**Reply by drjfloyd** [reply]
This doesn't correspond well to the assumptions made in a zone model.

**Reply by SXFire12** [reply]
do you have any suggestion for that case? or if we average the upper layer temperature of the two compartments, will the final result be reasonable?

**Reply by drjfloyd** [reply]
model as one compartment.
